src/fipp/edn.cljc

```diff
@@ -19,10 +19,26 @@
                    [:span sep "..."])]
     [:group open [:align ys ellipsis] close]))
 
-(defrecord EdnPrinter [symbols print-meta print-length print-level]
+(defn- cache-satisfies
+  ([prot]
+   (let [cache (volatile! {})]
+     (fn [x]
+       (cache-satisfies prot cache x))))
+  ([prot cache x]
+   (let [clazz (type x)
+         ret (get @cache clazz ::not-found)]
+     (if (identical? ::not-found ret)
+       (let [ret (satisfies? prot x)]
+         (vswap! cache assoc clazz ret)
```

Comment:
Since the protocol is not used in the cache itself, it's misleading to pass it in. Should instead just specialize to IOverride and drop the parameter.

Reply:
Just so I don't misunderstand, drop the protocol parameter, yes?

Reply:
Right